src/state/hooks.ts

```diff
@@ -142,17 +142,33 @@ export const useCakeVault = () => {
     },
   } = useSelector((state: State) => state.pools.cakeVault)
 
-  // TODO: Figure out why this needs to be memoised to prevent rerender loop within BountyCard effect
   const estimatedCakeBountyReward = useMemo(() => {
     return new BigNumber(estimatedCakeBountyRewardAsString)
   }, [estimatedCakeBountyRewardAsString])
 
-  const totalShares = new BigNumber(totalSharesAsString)
-  const pricePerFullShare = new BigNumber(pricePerFullShareAsString)
-  const totalPendingCakeHarvest = new BigNumber(totalPendingCakeHarvestAsString)
-  const totalCakeInVault = new BigNumber(totalCakeInVaultAsString)
-  const userShares = new BigNumber(userSharesAsString)
-  const cakeAtLastUserAction = new BigNumber(cakeAtLastUserActionAsString)
+  const totalPendingCakeHarvest = useMemo(() => {
+    return new BigNumber(totalPendingCakeHarvestAsString)
+  }, [totalPendingCakeHarvestAsString])
+
+  const totalShares = useMemo(() => {
```

Comment:
Memoise so that if the BigNumbers returned by `useCakeVault` are used in dependency arrays - they don't trigger re-renders